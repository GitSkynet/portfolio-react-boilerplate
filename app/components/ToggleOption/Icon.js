import PropTypes from 'prop-types';
import styled from 'styled-components';
import { rem } from 'polished';

const Icon = styled.div`
  align-items: center;
  display: flex;
  height: ${rem(24)};
  width: ${rem(36)};
  border: 1px solid #575756;

  > img {
    filter: ${(props) => props.isChecked ? 'saturate(150%)' : 'saturate(50%)'};
  }

  &:hover > img {
    filter: saturate(150%);
  }
`;

Icon.propTypes = {
  isChecked: PropTypes.bool,
};

export default Icon;
