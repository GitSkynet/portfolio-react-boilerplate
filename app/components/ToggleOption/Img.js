import styled from 'styled-components';

import NormalImg from '../../components/Img';

const Img = styled(NormalImg)`
  filter: saturate(100%);
  transition: all 0.2s;
`;

export default Img;
