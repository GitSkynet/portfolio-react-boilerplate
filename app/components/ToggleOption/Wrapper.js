import styled from 'styled-components';
import { rem } from 'polished';

const Wrapper = styled.div`
  padding: ${rem(7)};
  background-color: #797978;
`;
export default Wrapper;
