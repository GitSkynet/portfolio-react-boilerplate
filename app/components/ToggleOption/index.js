/**
*
* ToggleOption
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import Icon from './Icon';
import Img from './Img';
import Wrapper from './Wrapper';

const ToggleOption = ({ value, message, onChange, isChecked, iconImg }) => (
  <Wrapper>
    <label htmlFor={`js-radio-label-${value}`}>
      <Icon
        isChecked={isChecked}
        aria-hidden
      ><Img
        src={iconImg}
        alt={''}
      /></Icon><span hidden aria-hidden={false}>{
      message
        ? <FormattedMessage {...message} />
        : <span>{value}</span>
    }</span>
    </label>
    <input
      hidden
      id={`js-radio-label-${value}`}
      type={'radio'}
      name={'localeToggle'}
      value={value}
      onChange={onChange}
      checked={isChecked}
    />
  </Wrapper>
);

ToggleOption.propTypes = {
  value: PropTypes.string.isRequired,
  message: PropTypes.object,
  onChange: PropTypes.func,
  iconImg: PropTypes.string,
  isChecked: PropTypes.bool,
};

export default ToggleOption;
