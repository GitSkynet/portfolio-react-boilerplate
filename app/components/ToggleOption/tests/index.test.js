import React from 'react';
import { shallow, mount } from 'enzyme';
import { IntlProvider, defineMessages } from 'react-intl';

import ToggleOption from '../index';

describe('<ToggleOption />', () => {
  it('should render and match the snapshot', () => {
    const defaultEnMessage = 'someContent';
    const message = defineMessages({
      enMessage: {
        id: 'boilerplate.containers.LocaleToggle.en',
        defaultMessage: defaultEnMessage,
      },
    });
    const onChange = jest.fn();
    const iconImg = './image.png';
    const renderedComponent = shallow(
      <IntlProvider locale="en">
        <ToggleOption
          value="en"
          message={message.enMessage}
          onChange={onChange}
          iconImg={iconImg}
        />
      </IntlProvider>
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render default language messages', () => {
    const defaultEnMessage = 'someContent';
    const message = defineMessages({
      enMessage: {
        id: 'boilerplate.containers.LocaleToggle.en',
        defaultMessage: defaultEnMessage,
      },
    });
    const renderedComponent = shallow(
      <IntlProvider locale="en">
        <ToggleOption value="en" message={message.enMessage} />
      </IntlProvider>
    );
    expect(renderedComponent.contains(<ToggleOption value="en" message={message.enMessage} />)).toBe(true);
  });

  it('should display `value`(two letter language code) when `message` is absent', () => {
    const renderedComponent = mount(
      <IntlProvider locale="nl">
        <ToggleOption value="nl" />
      </IntlProvider>
    );
    expect(renderedComponent.text()).toBe('nl');
  });
});
