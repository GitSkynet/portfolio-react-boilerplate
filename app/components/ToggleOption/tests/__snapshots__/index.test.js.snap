// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleOption /> should render and match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IntlProvider
    locale="en"
>
    <ToggleOption
        iconImg="./image.png"
        message={
            Object {
                "defaultMessage": "someContent",
                "id": "boilerplate.containers.LocaleToggle.en",
              }
        }
        onChange={[Function]}
        value="en"
    />
</IntlProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "iconImg": "./image.png",
      "message": Object {
        "defaultMessage": "someContent",
        "id": "boilerplate.containers.LocaleToggle.en",
      },
      "onChange": [Function],
      "value": "en",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "iconImg": "./image.png",
        "message": Object {
          "defaultMessage": "someContent",
          "id": "boilerplate.containers.LocaleToggle.en",
        },
        "onChange": [Function],
        "value": "en",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
